pytket/tests/classical_test.py

```diff
@@ -481,6 +483,164 @@ def test_add_wasm_to_reg() -> None:
     assert c.depth() == 4
 
 
+def test_wasm_argtypes() -> None:
+    w = wasm.WasmFileHandler("testfile.wasm")
+
+    with open("testfile.wasm", "rb") as f:
+        bytecode = f.read()
+    assert w.bytecode() == bytecode
+
+    c = Circuit(0, 6)
+
+    c.add_wasm("add_one", w, [1], [1], [Bit(0), Bit(1)])
+
+    assert c.depth() == 1
+
+    cmd = c.get_commands()[0]
+
+    assert isinstance(cmd.args[0], unit_id.Bit)
+    assert isinstance(cmd.args[1], unit_id.Bit)
+    assert isinstance(cmd.args[2], unit_id.WasmState)
+
+
+def test_wasm_uid_from_circuit() -> None:
+    w = wasm.WasmFileHandler("testfile.wasm")
+
+    with open("testfile.wasm", "rb") as f:
+        bytecode = f.read()
+    assert w.bytecode() == bytecode
+
+    c = Circuit(0, 6)
+
+    c.add_wasm("add_one", w, [1], [1], [Bit(0), Bit(1)])
+
+    assert c.depth() == 1
+
+    assert (
+        c.wasm_uid == "6a0a29e235cd5c60353254bc2b459e631d381cdd0bded7ae6cb44afb784bd2de"
+    )
+
+
+def test_wasm_uid_from_circuit_2() -> None:
+    w = wasm.WasmFileHandler("testfile.wasm")
+
+    with open("testfile.wasm", "rb") as f:
+        bytecode = f.read()
+    assert w.bytecode() == bytecode
+
+    c = Circuit(0, 6)
+
+    c.add_wasm("add_one", w, [1], [1], [Bit(0), Bit(1)], condition=Bit(3))
+
+    assert c.depth() == 1
+
+    assert (
+        c.wasm_uid == "6a0a29e235cd5c60353254bc2b459e631d381cdd0bded7ae6cb44afb784bd2de"
+    )
+
+
+def test_wasm_uid_from_circuit_3() -> None:
+    wfh = wasm.WasmFileHandler("testfile.wasm")
+
+    c = Circuit(0)
+    a = c.add_c_register("c", 8)
+    c.add_wasm_to_reg("add_one", wfh, [a], [a])
+    assert c.depth() == 1
+
+    cbox = CircBox(c)
+    d = Circuit(0, 8)
+
+    d.add_circbox(cbox, [0, 1, 2, 3, 4, 5, 6, 7])
+
+    assert (
+        d.wasm_uid == "6a0a29e235cd5c60353254bc2b459e631d381cdd0bded7ae6cb44afb784bd2de"
+    )
+
+
+def test_wasm_uid_from_circuit_4() -> None:
+    wfh = wasm.WasmFileHandler("testfile.wasm")
+
+    c = Circuit(0)
+    a = c.add_c_register("c", 8)
+    c.add_wasm_to_reg("add_one", wfh, [a], [a])
+    assert c.depth() == 1
+
+    cbox = CircBox(c)
+    d = Circuit(0, 8)
+
+    d.add_circbox(cbox, [0, 1, 2, 3, 4, 5, 6, 7], condition=Bit(0))
+
+    assert (
+        d.wasm_uid == "6a0a29e235cd5c60353254bc2b459e631d381cdd0bded7ae6cb44afb784bd2de"
+    )
+
+
+def test_wasm_uid_from_circuit_5() -> None:
+    w = wasm.WasmFileHandler("testfile.wasm")
+
+    with open("testfile.wasm", "rb") as f:
+        bytecode = f.read()
+    assert w.bytecode() == bytecode
+
+    c = Circuit(0, 6)
+
+    assert c.depth() == 0
+    with pytest.raises(ValueError):
+        c.wasm_uid
```

Comment:
I think it would be a bit nicer if this returned `None` when there is no WASM wire, rather than throw an exception.